### PR TITLE
Route custom-review base refs through branch-diff packets

### LINKED ISSUE #177: Kimi CLI fails to produce review verdict on tiny packets and no-source continuations
## Problem

Kimi remains an unreliable external-review slot after the #171 shared source-packet/no-resend policy changes. The latest PR #175 review on head `3b62737` used a tiny docs-only branch-diff packet and still failed to produce a usable verdict.

Current evidence:

- PR: #175
- Scope: branch-diff `38d4c59..3b62737`
- Selected source: 2 Markdown files, 15,381 bytes, 149 lines
- Initial Kimi job: `182032de-e7da-4c1f-836d-eefdc8f01e61`
  - Source state: sent
  - Failure: `step_limit_exceeded`
  - Message: `Max number of steps reached: 96`
- First no-source continuation: `c70ce850-5eb5-4b40-b175-0130c64658a5`
  - Selected source: zero files, zero bytes
  - Source state: not_sent
  - Failure: `step_limit_exceeded`
  - Message: `Max number of steps reached: 160`
- Second no-source continuation: `0ff438dd-2613-4a03-8a97-46b0427e1e89`
  - Selected source: zero files, zero bytes
  - Source state: not_sent
  - Failure: `timeout`
  - Message: `target CLI exceeded the configured timeoutMs`
  - Step budget: 320

This proves the remaining symptom is not just large source packets. Kimi can fail or time out even when source is tiny and retry policy correctly avoids resending source.

## Expected Outcome

- Kimi returns a usable verdict for small branch-diff review packets or fails quickly with deterministic, source-safe diagnostics.
- No-source continuation should complete or fail with a clear bounded reason before exhausting large step budgets/timeouts.
- Failed Kimi slots remain failed slots and never count as approval.
- If Kimi needs provider-specific step/timeout/model limits, those limits are recorded as adapter capability facts under the shared policy.

## Approach Notes

- Keep #171 shared policy intact; do not add Kimi-only hidden behavior.
- Investigate Kimi CLI runtime behavior, prompt shape, model choice, timeout defaults, and whether the companion should use smaller verdict-only repair packets.
- If Kimi direct API or OpenRouter support becomes available, it must enter the same shared route ladder as other providers.
- Do not automatically resend selected source after source-bearing failure.

## Risks

- Six-provider review gates remain blocked or require operator waiver.
- Tiny packets can still consume Kimi runtime without producing a verdict.
- Operators may confuse a source-sent attempt with a usable approval unless failed-slot semantics remain strict.

## Verification

- Add a minimal branch-diff fixture that Kimi should review with one verdict line.
- Add no-source continuation tests proving zero selected files, `source_content_transmission=not_sent`, and bounded failure semantics.
- Run Kimi review manually against a small docs-only delta and confirm either `Verdict: APPROVE` or a deterministic failed slot without source resend.

## Relations

Related to #171
Related to #172

### PR DESCRIPTION
**Pull request by seungpyoson**
Closes #177.

## Problem and Root Cause
Kimi #177 no longer points at auth/readiness as the primary blocker: live issue evidence shows source-free Kimi readiness is OK and under-budget branch-diff shards can complete after the account upgrade. The remaining failure class is packet shape. Explicit-path `custom-review` remained classified as `custom` even when the operator supplied `--scope-base <ref>`, preserving the full-file custom-scope path that can exceed provider packet budgets.

The root cause was shared profile normalization: `effectiveProfileForOptions` only converted `review --scope-base ...` to `branch-diff`; `custom-review --scope-paths ... --scope-base ...` stayed `custom`.

## Why This Is Provider-Neutral
This is not a Kimi-only shortcut. The behavior lives in shared `companion-common.mjs` and is synced to Claude, Gemini, and Kimi packaged copies. Provider-specific Kimi facts from #175 remain capability facts only: Kimi has a 32 KiB source-packet cap and unsupported no-source repair after source-bearing step-limit failures. Shared policy now lets committed explicit-path shards use branch-diff packets when a base ref is supplied.

## Implementation Summary
- Treat `custom-review --scope-base <ref>` as effective `branch-diff` scope while preserving explicit `--scope-paths` as the shard filter.
- Keep `custom-review` without `--scope-base` as explicit full-file custom scope for non-git bundles or intentional file-body review.
- Added #177 evidence map under `specs/177-kimi-review-verdict-reliability/`.
- Added shared helper/unit coverage and Kimi smoke coverage proving a large full file with a tiny committed diff sends the branch-diff packet, excludes a full-file-only sentinel, cleans up `runCompanion` data dir, and stays below Kimi's 32 KiB cap.

## Verification
Head: `848fc665d27cb0eef7dfd999bb502b3c256b207f`
Base: `origin/main` / `3cd27ef3c095c64009829ae4e9b6a3da51e65bc5`

Fresh local verification on clean detached exact-head worktree:
- `npm ci` - pass.
- `git diff --check origin/main...HEAD` - pass.
- `npm run lint:sync` - pass.
- `node --test --test-name-pattern "kimi custom-review with scope-base uses branch-diff packet|kimi review --scope-base preserves branch-diff scope" tests/smoke/kimi-companion.smoke.test.mjs` - pass 2/2.
- `node --test tests/unit/companion-common.test.mjs tests/smoke/kimi-companion.smoke.test.mjs tests/unit/plugin-copies-in-sync.test.mjs` - pass 168/168.

Remote CI:
- GitHub Actions `pull-request-ci` run `26431756565` - success.
- Jobs green: `lint`, `test`, `smoke (api-reviewers)`, `smoke (claude)`, `smoke (gemini)`, `smoke (grok)`, `smoke (kimi)`.
- SonarQube quality gate passed with 0 new issues.

External reviews at exact head `848fc66`, whole PR scope `origin/main...HEAD`:
- Gemini adversarial review - APPROVE.
- Claude adversarial review - APPROVE.
- Kimi adversarial review - APPROVE.
- Grok adversarial review - APPROVE.
- DeepSeek adversarial review - APPROVE.
- GLM adversarial review - APPROVE.

Review finding verification:
- Gemini Code Assist temp-dir cleanup finding was valid and fixed in `848fc66`.
- DeepSeek's note about missing DeepSeek/GLM `companion-common.mjs` copies was checked and is not valid: API reviewers do not package `companion-common.mjs`.
- DeepSeek's option-key-shape concern was checked and is not a bug: companion `parseArgs` stores `--scope-base` as `options["scope-base"]`, and the Kimi smoke test exercises the real CLI path.
- Whitespace-only custom-review `scope-base` is a minor test-parity note, not a behavior gap: `scopeBaseForOptions` tests whitespace handling directly, and both `review` and `custom-review` flow through that same helper.

## Remaining Scope
#176 Grok readiness/transport remains separate and should be handled in its own PR after #177 merges or is explicitly paused.

### CODE REVIEW

File: tests/smoke/kimi-companion.smoke.test.mjs [hunk 1]
```diff
@@ -2784,6 +2784,61 @@ test("kimi review --scope-base preserves branch-diff scope through target execut
...
+    assert.equal(manifest.scope_resolution.scope, "branch-diff");
+    assert.deepEqual(manifest.scope_resolution.scope_paths, ["doc.md"]);
+    assert.deepEqual(manifest.selected_source.files.map((file) => file.path), ["doc.md"]);
+    assert.equal(manifest.source_packet_policy.source_packet_action, "send");
+    assert.equal(manifest.source_packet_policy.source_send_allowed, true);
+    assert.equal(manifest.source_packet_policy.selected_source_bytes < 32 * 1024, true);
+  } finally {
+    rmSync(cwd, { recursive: true, force: true });
+  }
+});
```
[the remark this answers is not in the document]
Comment (seungpyoson) [reply tests/smoke/kimi-companion.smoke.test.mjs:2843]: Verified valid and fixed in `848fc665d27cb0eef7dfd999bb502b3c256b207f`: the new smoke test now tracks `result.dataDir` and removes it in `finally`. Current-head focused test is green (`2/2`), broader Kimi/unit/plugin-copy set is green (`168/168`), and remote CI is green.
